packages/tools/fluid-runner/src/logger/baseFileLogger.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as fs from "fs";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { IFileLogger } from "./fileLogger";
+
+/**
+ * @internal
+ */
+export abstract class BaseFileLogger implements IFileLogger {
+    public supportsTags?: true | undefined;
+
+    /** Hold events in memory until flushed */
+    protected events: string[] = [];
+    protected hasWrittenToFile = false;
+
+    /**
+     * @param filePath - file path to write logs to
+     * @param eventsPerFlush - number of events per flush
+     * @param defaultProps - default properties to add to every telemetry event
+     */
+     public constructor(
+        protected readonly filePath: string,
+        protected readonly eventsPerFlush: number = 50,
+        protected readonly defaultProps?: Record<string, string>,
+    ) { }
+
+    public send(event: ITelemetryBaseEvent): void {
+        // eslint-disable-next-line no-param-reassign
+        event = { ...event, ...this.defaultProps };
+        const logEvent = JSON.stringify(event);
+        this.events.push(logEvent);
+
+        if (this.events.length >= this.eventsPerFlush || event.category === "error") {
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            this.flush();
+        }
+    }
+
+    protected abstract flush(): Promise<void>;
+
+    protected async flushCore(outputPath: string, delimitter: string): Promise<void> {
```

Comment:
Not sure you need flushCore() - you can just rename it to be flush() and let CSVFileLogger overwrite it and call super.flush(). The only benefit of this structure I can see (if we want it) - it forces derived classes to do something, i.e., it brings attention to this area.